### PR TITLE
Use separate sessions in boto3 requests

### PR DESCRIPTION
**Pull request by agrawalsourav98**
Use separate sessions for boto3 requests instead of the default session to ensure threadsafety across s3 operations. This allows to ues multithreaded s3 operations downstream.

### CODE REVIEW

**Comment by agrawalsourav98** [comment]
All the class parameter changes to `None` are related to `B006` flake8 error from the [flake8-bugbear](https://pypi.org/project/flake8-bugbear/) extension.
```
B006: Do not use mutable data structures for argument defaults. They are created during function definition time.
All calls to the function reuse this one instance of that data structure, persisting changes between them.
```
The above changes are a better way to represent the same logic.

**Comment by isaaccorley** [comment]
I will wait for Conor's review on this so he's aware of the changes to the classes args since I believe it's used in some detection code elsewhere.

File: ml_dronebase_data_utils/s3/s3.py [hunk 1]
```diff
@@ -16,6 +18,31 @@
 from tqdm import tqdm
 
 
+def chunk(it: Iterable, size: int) -> Iterator:
```
Comment (isaaccorley) [inline ml_dronebase_data_utils/s3/s3.py:21]: Is there not one of our libraries that already implements this?

**Review by isaaccorley — CHANGES_REQUESTED** [review]
Can you make sure the tests cover these so we can verify that the changes works?

File: tests/test_s3.py
Comment (agrawalsourav98) [inline tests/test_s3.py:1]: Add test cases to test all the s3 functions.

File: ml_dronebase_data_utils/s3/s3.py (hunk 1 above)
Comment (isaaccorley) [reply ml_dronebase_data_utils/s3/s3.py:24]: I was thinking itertools might have this but maybe it doesn't.

**Review by isaaccorley — APPROVED** [review]
Looks great!

**Pull request merged**